versionist.conf.js

```diff
@@ -1,32 +1,59 @@
-module.exports = {
+var _ = require('lodash');
+var execSync = require('child_process').execSync;
 
-  subjectParser: 'angular',
+var getAuthor = (commitHash) => {
+  return execSync(`git show --quiet --format="%an" ${commitHash}`, { encoding: 'utf8' }).replace('\n', '');
+}
 
+module.exports = {
+  // This setup allows the editing and parsing of footer tags to get version and type information,
+  // as well as ensuring tags of the type 'v<major>.<minor>.<patch>' are used.
+  // It increments in a semver compatible fashion and allows the updating of NPM package info.
+  editChangelog: true,
+  parseFooterTags: true,
   getGitReferenceFromVersion: 'v-prefix',
+  incrementVersion: 'semver',
+  updateVersion: 'npm',
 
+  // Always add the entry to the top of the Changelog, below the header.
   addEntryToChangelog: {
     preset: 'prepend',
-    fromLine: 5
+    fromLine: 6
   },
 
-  updateVersion: 'npm',
-
+  // Only include a commit when there is a footer of 'change-type'.
+  // Ensures commits which do not up versions are not included.
+  // Commit messages without a relevant footer will not be included in the CHANGELOG.
   includeCommitWhen: (commit) => {
-    return commit.footer['changelog-entry'];
+    return !!commit.footer['change-type'] || !!commit.footer['changelog-entry'];
   },
 
+  // Determine the type from 'change-type:' tag.
+  // Should no explicit change type be made, then no changes are assumed.
   getIncrementLevelFromCommit: (commit) => {
-    return commit.footer['change-type'];
+    return _.trim(_.get(commit.footer, 'change-type', '')) || undefined;
+  },
+
+  // If a 'changelog-entry' tag is found, use this as the subject rather than the
+  // first line of the commit.
+  transformTemplateData: (data) => {
+    data.commits.forEach((commit) => {
+      commit.subject = commit.footer['changelog-entry'] || commit.subject;
+      commit.author = getAuthor(commit.hash);
+    });
+
+    return data;
   },
 
   template: [
-    '## {{version}} - {{moment date "Y-MM-DD"}}',
+    '# v{{version}} - {{moment date "Y-MM-DD"}}',
```

Comment:
Just out of curiosity, what's the reason for the change in the number of `#` characters? (i.e. the header-level)